lib/document.js

```diff
@@ -791,14 +779,7 @@ Document.prototype._saveSelf = function(
     }, reject);
   }
 
-  if (self.__proto__._model._tableReady === true) {
-    saveSelfHelper();
-  }
-  else {
-    model.once('ready', function() {
-      saveSelfHelper();
-    })
-  }
+  self.constructor.ready().then(saveSelfHelper);
```

Comment:
`saveSelfHelper` was created to avoid redundant code. Now that there's `ready`, there's no need to create a function in the middle of a method.
If you don't have time no worries, it's just more cleaning and we can fix that later in a later PR.